### PR TITLE
Build/conda using dependencies from conda

### PR DESCRIPTION
**Pull request by DerThorsten**
#318 but as a branch instead of fork

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,8 @@ project(arcticdb VERSION 0.0.1)
 
 enable_testing()
 
+option(ARCTICDB_USING_CONDA OFF)
```
Comment (jjerphan) [inline cpp/CMakeLists.txt:9]:
```suggestion
option(
    ARCTICDB_USING_CONDA
    "Whether ArcticDB's build system relies on conda for resolving its dependencies."
    OFF
)
```

**Review by jjerphan — APPROVED** [review]
LGTM modulo minor changes.

A few PRs have extracted changes from this one to make this PR slimmer.

Also https://github.com/man-group/ArcticDB/pull/352 has been opened to suggest some changes.

File: cpp/CMakePresets.json [hunk 2]
```diff
@@ -9,6 +9,19 @@
...
+      "name": "common_conda",
+      "hidden": true,
+      "description": "OS-agnostic config",
+      "binaryDir": "${sourceDir}/out/${presetName}-build",
+      "installDir": "${sourceDir}/out/install",
+      "cacheVariables": {
+        "ARCTICDB_USING_CONDA": "ON",
+        "ARCTICDB_USING_VCPKG": "OFF",
+        "STATIC_LINK_STD_LIB": "OFF",
+        "HIDE_LINKED_SYMBOLS": "OFF"
```
Comment (qc00) [inline cpp/CMakePresets.json:22]: I believe these are out of sync with another PR. For example, `ARCTICDB_USING_VCPKG` is gone now.

Also, can `STATIC_LINK_STD_LIB` and `HIDE_LINKED_SYMBOLS` be driven off `ARCTICDB_USING_CONDA`? I really want to simplify the profiles at some point.

File: cpp/CMakePresets.json [hunk 3]
```diff
@@ -68,12 +145,87 @@
...
+    {"name": "windows-cl-debug",          "configurePreset": "windows-cl-debug",          "targets": "arcticdb_ext" },
+    {"name": "windows-cl-release",        "configurePreset": "windows-cl-release",        "targets": "arcticdb_ext" },
+    {"name": "windows-cl-conda-debug",    "configurePreset": "windows-cl-conda-debug",    "targets": "arcticdb_ext" },
+    {"name": "windows-cl-conda-release",  "configurePreset": "windows-cl-conda-release",  "targets": "arcticdb_ext" },
+    {"name": "linux-debug",               "configurePreset": "linux-debug",               "targets": "arcticdb_ext" },
+    {"name": "linux-release",             "configurePreset": "linux-release",             "targets": "arcticdb_ext" },
+    {"name": "linux-conda-debug",        "configurePreset": "linux-conda-debug",          "targets": "arcticdb_ext" },
+    {"name": "linux-conda-release",      "configurePreset": "linux-conda-release",        "targets": "arcticdb_ext" },
+    {"name": "darwin-conda-debug",       "configurePreset": "darwin-conda-debug",         "targets": "arcticdb_ext" },
+    {"name": "darwin-codda-release",     "configurePreset": "darwin-conda-release",       "targets": "arcticdb_ext" }
```
Comment (qc00) [inline cpp/CMakePresets.json:229]: `configurePreset` is misaligned.

File: cpp/CMake/FindxxHash.cmake [hunk 4]
```diff
@@ -7,7 +7,7 @@
 # xxHash_VERSION_PATCH  - The patch version of xxHash
 
 find_path(xxHash_INCLUDE_DIR NAME xxhash.h PATH_SUFFIXES include)
-find_library(xxHash_LIBRARY NAMES libxxhash.a xxhash xxHash PATH_SUFFIXES lib)
+find_library(xxHash_LIBRARY NAMES libxxhash xxhash xxHash PATH_SUFFIXES lib)
```
[the remark this answers is not in the document]
Comment (jjerphan) [reply cpp/CMake/FindxxHash.cmake]: This is addressed separately in https://github.com/man-group/ArcticDB/pull/376, which completes #346.

File: cpp/CMakePresets.json (hunk 2 above)
Comment (jjerphan) [reply cpp/CMakePresets.json:22]: > I believe these are out of sync with another PR.

Which PR are you thinking about? Are you thinking about https://github.com/man-group/ArcticDB/pull/352?

**Comment by jjerphan** [comment]
@qc00: I let you merge when this LGTY. :slightly_smiling_face:

**Comment by DerThorsten** [comment]
once the CI is green I am fine with having it merged

**Review by mehertz — APPROVED** [review]

File: cpp/CMakePresets.json [hunk 5]
```diff
@@ -44,7 +64,51 @@
         "CMAKE_BUILD_TYPE": "Debug"
       }
     },
-
+    {
+      "name": "windows-cl-conda-release",
```
Comment (qc00) [inline cpp/CMakePresets.json:68]: I believe these should not be included in this PR given critical dependencies are not available via conda on Windows
